### PR TITLE
Fixes VSTS User Story 980492: Create unit test suite for

### PR DESCRIPTION
**Pull request by mkrueger**
DocumentRegistry

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/980492

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DocumentRegistry.cs [hunk 1]
```diff
@@ -53,28 +54,32 @@ static DocumentRegistry ()
 			FileService.FileChanged += HandleFileServiceChange;
 		}
 
-		static void HandleFileServiceChange (object sender, FileEventArgs e)
+		static async void HandleFileServiceChange (object sender, FileEventArgs e)
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DocumentRegistry.cs]: We should avoid async void.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DocumentRegistry.cs]: This changes the event handler to exit invoke when the first await is reached, which changes logic and does not ensure that all documents have been notified.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DocumentRegistry.cs]: The old way doesn't ensure it either. I don't want to change the event.
The biggest reason against async void is the exception handling which I handled here.

**Comment by Therzok** [comment]
@mkrueger the old way would exit only after _all_ the documents were triggered for reload. This exits as soon as the _first_ one is triggered for reload.

**Comment by mkrueger** [comment]
In the old way it may never be reloaded now it does. This implementation is more correct.

What do you mean with exit after the first one triggered ? the loop isn't canceled after reload. it exists after all have really reloaded.

File: main/tests/Ide.Tests/MonoDevelop.Ide.Gui.Documents/DocumentRegistryTests.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		[SetUp]
+		public async Task Setup ()
+		{
+			documentManager = await Runtime.GetService<DocumentManager> ();
+			documentControllerService = await Runtime.GetService<DocumentControllerService> ();
+			shell = (MockShell)await Runtime.GetService<IShell> ();
+		}
+
+		[Test]
+		public async Task TestVSTSBug8608 ()
```
Comment (jtorres) [inline main/tests/Ide.Tests/MonoDevelop.Ide.Gui.Documents/DocumentRegistryTests.cs:59]: Could we add a description here?

**Comment by mkrueger** [comment]
moved to vs for mac.

**Pull request closed without merging**